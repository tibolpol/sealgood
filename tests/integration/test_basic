#!/bin/bash

#######################################
# <$1                   : LINENO      #
# <stdin                : input       #
# <../samples/$2        : reference   #
# <../../bin/sealgood   : binary      #
# <../../locale/{fr,es} : locale      #
# >$TMPDIR/$2           : output      #
# >stdout               : test result #
#######################################
mydif(){
  lineno=$1
  shift
  tee "$TMPDIR/$(basename $1)" |
  diff -a "$mydir/$1" - || {
    rc=$?
    printf $'KO %4d %s\n' $lineno "$*"
    echo "!cp $TMPDIR/$(basename $1) $mydir/$1"
    return $rc
  }
  printf $'ok %4d %s\n' $lineno "$*"
}
export -f mydif
mdstyle(){
  echo '```'$1
  cat
  echo '```'
}
export -f mdstyle
exec {fddebug}> >(stdbuf -oL sed -e 's/++[0-9]*++$/\n&/;p' -e 's/^/== /' >&2)
export fddebug

export mydir="$(cd "$(dirname "$0")";pwd)" myname="$(basename "$0")"
PATH="$(cd "$mydir/../../bin";pwd):$PATH"
export TMPDIR=$(mktemp -d)
exec {fdtty}</dev/null
export fdtty # neutralise read <&$fdtty
parallel "cd $mydir/../../locale/{}/LC_MESSAGES&& msgfmt -o sealgood.mo sealgood.po" ::: fr es pt

exec 1> >(tee $mydir/$myname.stdout)

########################################################################
# help                                                                 #
# locale/{{run/fr,data/},{run/es,data/},{run/us,data/},{run/pt,data/}} #
########################################################################
parallel "mydif $LINENO ../../docs/{}/help_{}.md < <(LANGUAGE={} sealgood help 2>&1 | mdstyle console)" ::: us es fr pt

####################################################################
# lookup/PEM pubkey                                                #
# date/PEM pubkey                                                  #
# get_payload                                                      #
# locale/{{run,data}/fr,{run,data}/es,{run,data}/pt,{run,data}/us} #
####################################################################
parallel "mydif $LINENO ../samples/test_date_{} < <((
LANGUAGE={} sealgood date < $mydir/../samples/pempub 2>/dev/null) 2>&1 | sed -e 's/\(^MIIV\).*/\1/' )
" ::: us es fr pt

#################################################################################################
# lookup/PEM pubkey                                                                             #
# clean/PEM pubkey                                                                              #
# locale/run/{fr\ locale/data/{fr,es,us},es\ locale/data/{fr,es,us},us\ locale/data/{fr,es,us}} #
#################################################################################################
parallel "mydif $LINENO ../samples/test_clean < <(
LANGUAGE={1} sealgood clean < $mydir/../samples/test_date_{2} 2>/dev/null)" ::: fr es us pt ::: fr es us pt

#################################################################################################
# lookup/file_or_url_list                                                                       #
# lookup/sealgood+PEM pubkey                                                                    #
# clean/PEM pubkey                                                                              #
# locale/run/{fr\ locale/data/{fr,es,us},es\ locale/data/{fr,es,us},us\ locale/data/{fr,es,us}} #
#################################################################################################
parallel --linebuffer "mydif $LINENO ../samples/test_clean_list_{} < <(( ls $mydir/../samples/test_date_{fr,es,us,pt} |
LANGUAGE={} sealgood clean | tar -zvxOf-) 2>&1)" ::: fr es us pt

###################################
# lookup/file_or_url_list         #
# lookup/sealgood+PEM pubkey      #
# lookup/gzip                     #
# date/sealgood+PEM pubkey        #
# clean/sealgood+PEM pubkey       #
# verify/sealgood+PEM pubkey      #
# locale/{run/es,data/{fr,es,us}} #
###################################
mydif $LINENO ../samples/test_date_pubkey  < <(( ls $mydir/../samples/test_date_{es,fr,pt} |
LANGUAGE=es sealgood date verify |
                      tee $TMPDIR/pubkeys.tgz | # garde pour le test suivant
                      zcat|strings -n12 -weS) 2>&1 | sed -e 's/\(^MIIV\).*/\1/' -e 's/\(^Time stamp:\).*/\1/' )

############################
# lookup/tar+gzip          #
# date/sealgood+tar+gzip #
# lookup/sealgood+tar+gzip #
# verify/sealgood+tar+gzip #
# locale/{run,data}/us     #
############################
mydif $LINENO ../samples/test_date_gzip  < <(( cat $TMPDIR/pubkeys.tgz |
LANGUAGE=us sealgood date verify |
                      zcat|strings -n12 -weS) 2>&1 | sed -e 's/\(^MIIV\).*/\1/' -e 's/\(^Time stamp:\).*/\1/' \
                      -e 's/\(^wc       :\)   .[^ ].*/\1/' )

###################################
# lookup/file_or_url_list         #
# $STOPFILE                       #
# locale/{run,data}/fr            #
###################################
parallel "mydif $LINENO ../samples/test_stopfile_{}  < <(ls $mydir/../samples/test_date_{fr,es,us,pt} |
LANGUAGE=fr STOPFILE=/stopfile sealgood {} 2>&1)" ::: clean date
