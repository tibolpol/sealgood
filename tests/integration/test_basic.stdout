KO test non effectué depuis les dernières modifications
